app/assets/javascripts/components/generic_object/custom-image.js

```diff
@@ -35,7 +35,9 @@ function customImageController($timeout) {
     var imageFile;
 
     if (angular.element('#generic_object_definition_image_file')[0].files.length === 0) {
-      add_flash(__("No file chosen."), 'error');
+      if (!vm.imageMissing) {
+        add_flash(__("No file chosen."), 'error');
```

Comment:
you may want to add an explicit id - `add_flash(__("No file chosen."), 'error', {id: 'image-missing'});`

that way, calling add_flash multiple times will never render the flash more than once

(that should also fix the problem that currently, you can still get double message if you add the image and remove again)

Reply:
Neat!